src/main/scala/scala/compat/java8/FutureConverters.scala

```diff
@@ -78,6 +79,8 @@ object FutureConverters {
   def toScala[T](cs: CompletionStage[T]): Future[T] = {
     cs match {
       case cf: CF[T] => cf.wrapped
+      case cf: CompletableFuture[T] if cf.isDone =>
+        Future fromTry Try(cf.get()).recoverWith { case e: ExecutionException => Failure(e.getCause) }
```

Comment:
Given this code it is not clear to me that this is in fact an optimization. Is there any benchmark performed that shows what improvement this change leads to?

Reply:
@viktorklang Might be I missing something, but I believe that optimised version works faster because of removal of possible context switch

Reply:
@t3hnar I suspect this to be slower: there will be one allocation for the thunk to Try.apply, then another thunk for the recoverWith function. For the other code it will be a single allocation of the P.